thirdparty/patchmatchnet/models/net.py

```diff
@@ -0,0 +1,365 @@
+
+        intra_feat = F.interpolate(intra_feat, scale_factor=2, mode="bilinear") + self.inner2(conv4)
+        del conv4
+        output_feature["stage_1"] = self.output3(intra_feat)
+
+        del intra_feat
+
+        return output_feature
+
+
+class Refinement(nn.Module):
+    """depth map refinement network"""
+
+    def __init__(self):
+        """ initialization method"""
+
+        super(Refinement, self).__init__()
+
+        # img: [B,3,H,W]
+        self.conv0 = ConvBnReLU(in_channels=3, out_channels=8)
+        # depth map:[B,1,H/2,W/2]
+        self.conv1 = ConvBnReLU(in_channels=1, out_channels=8)
+        self.conv2 = ConvBnReLU(in_channels=8, out_channels=8)
+        self.deconv = nn.ConvTranspose2d(
+            in_channels=8, out_channels=8, kernel_size=3, padding=1, output_padding=1, stride=2, bias=False
+        )
+
+        self.bn = nn.BatchNorm2d(8)
+        self.conv3 = ConvBnReLU(in_channels=16, out_channels=8)
+        self.res = nn.Conv2d(in_channels=8, out_channels=1, kernel_size=3, padding=1, bias=False)
+
+    def forward(
+        self, img: torch.Tensor, depth_0: torch.Tensor, depth_min: torch.Tensor, depth_max: torch.Tensor
+    ) -> torch.Tensor:
+        """forward methods
+        Args:
+            img: input source and reference images (B, 3, H, W)
+            depth_0: current depth map (B, 1, H//2, W//2)
+            depth_min: pre-defined minimum depth
+            depth_max: pre-defined maximum depth
+        Returns:
+            depth: refined depth map (B, 1, H//2, W//2)
+        """
+
+        batch_size = depth_min.size()[0]
+        # pre-scale the depth map into [0,1]
+        depth = (depth_0 - depth_min.view(batch_size, 1, 1, 1)) / (
+            depth_max.view(batch_size, 1, 1, 1) - depth_min.view(batch_size, 1, 1, 1)
+        )
+
+        conv0 = self.conv0(img)
+        deconv = F.relu(self.bn(self.deconv(self.conv2(self.conv1(depth)))), inplace=True)
+        cat = torch.cat((deconv, conv0), dim=1)
+        del deconv, conv0
+        # depth residual
+        res = self.res(self.conv3(cat))
+        del cat
+
+        depth = F.interpolate(depth, scale_factor=2, mode="nearest") + res
+        # convert the normalized depth back
+        depth = depth * (depth_max.view(batch_size, 1, 1, 1) - depth_min.view(batch_size, 1, 1, 1)) + depth_min.view(
+            batch_size, 1, 1, 1
+        )
+
+        return depth
+
+
+class PatchmatchNet(nn.Module):
+    """ Implementation of complete structure of Patchmatch Net"""
+
+    def __init__(
+        self,
+        patchmatch_interval_scale: List[float] = [0.005, 0.0125, 0.025],
+        propagation_range: List[int] = [6, 4, 2],
+        patchmatch_iteration: List[int] = [1, 2, 2],
+        patchmatch_num_sample: List[int] = [8, 8, 16],
+        propagate_neighbors: List[int] = [0, 8, 16],
+        evaluate_neighbors: List[int] = [9, 9, 9],
+    ) -> None:
+        """initialize modulse in patchmatch net

```

Comment:
Small typo
"Initialize modules in PatchmatchNet"

prefer to capitalize the first word of the  1-line description